### ISSUE TITLE
Out Of Memory at application restart in Debug

### ISSUE THREAD

**Issue by devproivansurzhenko** [bugsnag/bugsnag-react-native#378]
### Description
I am experiencing OOM reports when my app is restarted locally from Xcode (i.e. if I am made some dev changes on RN side)
**It happens in 30-40% of restarts**

<img width="753" alt="Screen Shot 2019-07-10 at 4 23 47 PM" src="https://user-images.githubusercontent.com/48912267/60972794-a182b900-a32f-11e9-8b47-2302f8d436a7.png">
<img width="814" alt="Screen Shot 2019-07-10 at 4 24 10 PM" src="https://user-images.githubusercontent.com/48912267/60972795-a21b4f80-a32f-11e9-9048-72622cb38728.png">
<img width="561" alt="Screen Shot 2019-07-10 at 4 24 03 PM" src="https://user-images.githubusercontent.com/48912267/60972797-a21b4f80-a32f-11e9-9b70-03f48cd8c0f0.png">
<img width="525" alt="Screen Shot 2019-07-10 at 4 23 53 PM" src="https://user-images.githubusercontent.com/48912267/60972798-a2b3e600-a32f-11e9-8e7b-d9fbd1b6cf9c.png">

### Issue
In fact, there is no OOM state. App is just restarted.

### Environment

Library versions:
```
"@react-native-community/async-storage": "^1.2.2",
    "@segment/analytics-react-native": "0.0.1-beta.5",
    "@types/lodash": "^4.14.120",
    "@types/react-native-indicators": "^0.13.1",
    "@types/react-native-vector-icons": "^4.6.4",
    "@types/react-native-video": "^3.1.1",
    "@types/react-navigation": "^3.0.1",
    "bugsnag-react-native": "^2.16.0",
    "libphonenumber-js": "^1.7.15",
    "lodash": "^4.17.11",
    "mock-async-storage": "^2.0.5",
    "moment": "^2.24.0",
    "native-base": "^2.12.0",
    "query-string": "^6.6.0",
    "react": "16.8.6",
    "react-native": "0.59.3",
    "react-native-contacts": "^4.0.0",
    "react-native-default-preference": "^1.3.2",
    "react-native-gesture-handler": "^1.0.15",
    "react-native-indicators": "^0.13.0",
    "react-native-keyboard-aware-scroll-view": "^0.8.0",
    "react-native-mail": "^3.0.7",
    "react-native-oauth": "^2.1.18",
    "react-native-phone-call": "^1.0.9",
    "react-native-status-bar-size": "^0.3.3",
    "react-native-swipe-list-view": "^1.5.1",
    "react-native-vector-icons": "^6.2.0",
    "react-native-version-number": "^0.3.5",
    "react-native-video": "^4.4.0",
    "react-navigation": "^3.2.1"
```

- cocoapods version (if any) (`pod -v`): 1.7.2
- iOS/Android version(s): iOS
- simulator/emulator or physical device?: both
- debug mode or production?: debug

- [ ] (iOS only) `[BugsnagReactNative start]` is present in the
  `application:didFinishLaunchingWithOptions:` method in your `AppDelegate`
  class?
No

### Example Repo

My steps:
1. Open ios project from Xcode
2. Change Run schema to `Debug` (also, enable `Debug executable` option)
3. Run the app

**Comment by mattdyoung** [comment]
Hi @devproivansurzhenko 

We've also seen this behaviour as Xcode terminates the existing app launch session with a kill signal when restarting an app in debug mode, which looks like the OS terminating the app to our detection logic.

We've resolved this on a PR which is due to be released shortly, by disabling out of memory reporting in debug mode:
https://github.com/bugsnag/bugsnag-cocoa/pull/380

**Comment by devproivansurzhenko** [comment]
@mattdyoung Thank you for keeping me up to date!

**Comment by mattdyoung** [comment]
Now released in v5.22.3:
https://github.com/bugsnag/bugsnag-cocoa/releases/tag/v5.22.3
